File: 189.py
class Solution:
  def rotateArray(self, k: int, nums: list[int]) -> list[int]:
    length = len(nums)
    start = length - k
    end = start + length
    nums = nums + nums
    return nums[start:end]


s = Solution()
assert s.rotateArray(3, [1, 2, 3, 4, 5, 6, 7]) == [5, 6, 7, 1, 2, 3, 4]
assert s.rotateArray(2, [-1, -100, 3, 99]) == [3, 99, -1, -100]

print("Python: success")

File: 189/189.pl
use 5.40.0;
use Test::More;
use Data::Printer;

sub solution {
  my ($k, $nums) = @_;
  my $length = @$nums;
  my $start = $length - $k;
  my $end = $start + $length - 1;
  my @nums = (@$nums, @$nums);
  # say "$start, $end";
  # p @nums;
  # p @nums[$start..$end];
  return [ @nums[$start..$end] ];
}

is_deeply(solution(3, [1,2,3,4,5,6,7]), [5,6,7,1,2,3,4]);
is_deeply(solution(2, [-1,-100,3,99]),  [3,99,-1,-100]);

done_testing;
